### PR TITLE
Add Querier|Scanner

### PR DESCRIPTION
**Pull request by xescugc**
This PR adds the default documentation of the usecases this lib fulfills

### CODE REVIEW

**Review by patrislav — COMMENTED** [review]
Would it make sense to include `golang/mock` mocks for the interfaces in this repository? Or do we expect the library consumers to generate them?

**Comment by xescugc** [comment]
> Would it make sense to include golang/mock mocks for the interfaces in this repository? Or do we expect the library consumers to generate them?

That is a really good question that I don't have any preference or strong opinion, I would say that the caller is the one to do it just because it may not be using `gomock`, so it's up to the caller/user to do it.

OFC if we where to use it on this lib then yes but it's not the case.

File: querier.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package sqlr
+
+import (
+	"context"
+	"database/sql"
+)
+
+// RowQuerier requests a row in specific context
```
Comment (codebien) [inline querier.go]:
```suggestion
// RowQuerier requests a row in a specific context
```
Executer has already this form so I suggest moving to it for the rest

**Comment by patrislav** [comment]
> That is a really good question that I don't have any preference or strong opinion, I would say that the caller is the one to do it just because it may not be using `gomock`, so it's up to the caller/user to do it.
> 
> OFC if we where to use it on this lib then yes but it's not the case.

I see the reason to not have it, still, it'd be very useful for a user of the library who wants to mock those interfaces. IMO it'd add value. But we can discuss it in another issue, this PR LGTM :+1:

**Comment by kerak19** [comment]
I think that the interface definition maybe should be shown along with the description.

So for example

## Scanner

Used to abstract the logic of Scanning for `sql.Row` and `sql.Rows`.
```go
type Scanner interface {
	Scan(dest ...interface{}) error
}
```
Example usage
```go
func scanUsers(rows *sql.Rows) ([]*user.User, error) {
    users := make([]*user.User, 0,0)
    defer rows.Close()
    for rows.Next() {
        u, err := scanUser(rows)
        if err != nil {
            return nil, err
        }
        users = append(users, u)
    }
    if err := rows.Err(); err != nil {
        return nil, err
    }
    return users, nil
}

// Here goes the sqlr.Scanner
func scanUser(s sqlr.Scanner) (*user.User,error) {
    var u user.User
    err := s.Scan(
        &u.Name,
    )
    if err != nil {
        return nil, err
    }
    return u, nil
}
```

Single example containing all the interfaces may be too long (for now it's only two, but it may change). Or maybe have separated examples showing only part with the type and one big for all of them at the end.

**Review by codebien — CHANGES_REQUESTED** [review]
Just a very small thing

**Comment by xescugc** [comment]
ok pushed the 2 changes.

**Comment by codebien** [comment]
LGTM RS when others say too

**Review by kerak19 — COMMENTED** [review]
Hmm, there's no example for `Querier`, but IDK if that's necessary.

The formatting of the code is not `Go`-ish, should we fix that?

**Comment by xescugc** [comment]
Done and RS

**Review by kerak19 — APPROVED** [review]

**Review by codebien — APPROVED** [review]

**Pull request merged**